### PR TITLE
Automate OBC creation for respective Noobaa Bucket

### PR DESCRIPTION
**Pull request by naveenpaul1**
- automatically create OBC for the Bucket class
- update Noobaa resource limit and request
- use PVPool for the Noobaa default bucket

Note : test will add in different PR

Signed-off-by: Naveen Paul

### CODE REVIEW

File: controllers/mcg_bucket.go [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+	DefaultBackingStore = "noobaa-default-backing-store"
+)
+
+func (r *ManagedMCGReconciler) watchBucketClass(object client.Object) {
+	bucketName := object.GetName()
+	annotations := object.GetAnnotations()
+	if _, ok := annotations[McgmsObcNamespace]; ok {
+		r.objectBucketClaim = &noobaav1alpha1.ObjectBucketClaim{}
+		r.bucketClass = &noobaav1alpha1.BucketClass{}
+		r.ctx = context.Background()
```
Comment (rexagod) [inline controllers/mcg_bucket.go:41]:
```suggestion
```
It seems that we are replacing the existing context here with a new one.

File: controllers/mcg_bucket.go [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+			return nil
+		})
+		if err != nil {
+			r.Log.Error(err, "Error while creating OBC, OBC not created")
+
+			return
+		}
+		annotations := object.GetAnnotations()
+		if isCacheEnabled, ok := annotations[McgmsCacheEnabled]; ok && isCacheEnabled == "true" {
```
Comment (rexagod) [inline controllers/mcg_bucket.go:61]: Who will add this annotation? The UI (as we talked earlier)?

**Review by rexagod — APPROVED** [review]
LGTM, just some minor non-blocking comments.

File: controllers/mcg_bucket.go (hunk 2 above)
Comment (naveenpaul1) [reply controllers/mcg_bucket.go:61]: @rexagod yes UI team

cc : @SanjalKatiyar

**Comment by naveenpaul1** [comment]
Created new PR with the same changes : https://github.com/red-hat-storage/mcg-osd-deployer/pull/37

**Pull request closed without merging**